src/main/kotlin/com/openlattice/search/renderers/AlprAlertEmailRenderer.kt

```diff
@@ -0,0 +1,188 @@
+import com.openlattice.search.PersistentSearchMessengerHelpers
+import com.openlattice.search.requests.PersistentSearch
+import jodd.mail.att.ByteArrayAttachment
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.net.URL
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+import javax.imageio.ImageIO
+
+private val logger = LoggerFactory.getLogger(AlprAlertEmailRenderer::class.java)
+
+private val FROM_EMAIL = "katherine@openlattice.com"
+private val TEMPLATE_PATH = "mail/templates/shared/AlprAlertTemplate.mustache"
+private val MIME_TYPE_PREFIX = "image/"
+private val JPEG = "jpeg"
+private val PNG = "png"
+
+private val LAST_READ_FQN = FullQualifiedName("ol.datetimelastreported")
+private val EXPIRATION_FQN = FullQualifiedName("ol.datetimeend")
+private val SEARCH_QUERY_FQN = FullQualifiedName("ol.searchquery")
+private val READ_TIMESTAMP_FQN = FullQualifiedName("ol.datelogged")
+private val PERSON_ID_FQN = FullQualifiedName("nc.SubjectIdentification")
+
+// alert details
+private val CASE_NUM_FQN = FullQualifiedName("criminaljustice.casenumber")
+private val SEARCH_REASON_FQN = FullQualifiedName("ol.searchreason")
+
+// vehicle details
+private val LICENSE_PLATE_IMAGE_FQN = FullQualifiedName("ol.licenseplateimage")
+private val VEHICLE_IMAGE_FQN = FullQualifiedName("ol.vehicleimage")
+private val LOCATION_COORDS_FQN = FullQualifiedName("ol.locationcoordinates")
+private val LICENSE_NUMBER_FQN = FullQualifiedName("vehicle.licensenumber")
+private val LICENSE_STATE_FQN = FullQualifiedName("vehicle.licensestate")
+private val MAKE_FQN = FullQualifiedName("vehicle.make")
+private val MODEL_FQN = FullQualifiedName("vehicle.model")
+private val YEAR_FQN = FullQualifiedName("vehicle.year")
+private val COLOR_FQN = FullQualifiedName("vehicle.color")
+private val AGENCY_NAME_FQN = FullQualifiedName("publicsafety.agencyname")
+private val CAMERA_ID_FQN = FullQualifiedName("ol.resourceid")
+private val HIT_TYPE_FQN = FullQualifiedName("ol.description")
+
+class AlprAlertEmailRenderer {
+
+    companion object {
+
+        private val dateTimeFormatter = DateTimeFormatter.ofPattern("MM/dd/yyyy hh:mm a, z").withZone(TimeZone.getTimeZone("America/Los_Angeles").toZoneId())
+
+        private fun getFirstValue(entity: SetMultimap<FullQualifiedName, Any>, fqn: FullQualifiedName): String? {
+            val values = entity.get(fqn)
+            return if (values.isEmpty()) {
+                null
+            } else values.iterator().next().toString()
+
+        }
+
+        private fun getImage(imageUrlPath: String?, fileType: String): ByteArrayOutputStream? {
+            if (imageUrlPath == null) {
+                return null
+            }
+
+            var stream: ByteArrayOutputStream? = null
+
+            try {
+                val imageUrl = URL(imageUrlPath)
```

Comment:
I'm sure this is working, but if you have binary data loaded from s3 (say using the aws s3 java sdk) and content type I think you can avoid lots of overhead here. Even if you don't fix it now file a future JIRA ticket to come back and fix this.